simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/component/connector/AvailabilityStateVariableAccessor.java

```diff
@@ -0,0 +1,134 @@
+import com.evbox.everon.ocpp.simulator.station.evse.Connector;
+import com.evbox.everon.ocpp.simulator.station.evse.Evse;
+import com.evbox.everon.ocpp.v20.message.centralserver.Component;
+import com.evbox.everon.ocpp.v20.message.centralserver.GetVariableResult;
+import com.evbox.everon.ocpp.v20.message.centralserver.SetVariableResult;
+import com.evbox.everon.ocpp.v20.message.centralserver.Variable;
+import com.evbox.everon.ocpp.v20.message.station.ReportDatum;
+import com.evbox.everon.ocpp.v20.message.station.StatusNotificationRequest;
+import com.evbox.everon.ocpp.v20.message.station.VariableAttribute;
+import com.evbox.everon.ocpp.v20.message.station.VariableCharacteristics;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static com.evbox.everon.ocpp.v20.message.station.VariableAttribute.Mutability.READ_ONLY;
+import static com.evbox.everon.ocpp.v20.message.station.VariableCharacteristics.DataType.SEQUENCE_LIST;
+import static java.util.Collections.singletonList;
+
+public class AvailabilityStateVariableAccessor extends VariableAccessor {
+
+    public static final String NAME = "AvailabilityState";
+    public static final String CONNECTOR_AVAILABILITY = "Available";
+
+    private final Map<AttributeType, VariableGetter> variableGetters = ImmutableMap.<AttributeType, VariableGetter>builder()
+            .put(AttributeType.ACTUAL, this::getActualValue)
+            .build();
+
+    private final Map<AttributeType, SetVariableValidator> variableValidators = ImmutableMap.<AttributeType, SetVariableValidator>builder()
+            .put(AttributeType.ACTUAL, this::rejectVariable)
+            .build();
+
+    public AvailabilityStateVariableAccessor(Station station, StationStore stationStore) {
+        super(station, stationStore);
+    }
+
+    @Override
+    public String getVariableName() {
+        return NAME;
+    }
+
+    @Override
+    public Map<AttributeType, VariableGetter> getVariableGetters() {
+        return variableGetters;
+    }
+
+    @Override
+    public Map<AttributeType, VariableSetter> getVariableSetters() {
+        return Collections.emptyMap();
+    }
+
+    @Override
+    public Map<AttributeType, SetVariableValidator> getVariableValidators() {
+        return variableValidators;
+    }
+
+    @Override
+    public List<ReportDatum> generateReportData(String componentName) {
+        List<ReportDatum> reportData = new ArrayList<>();
+
+        for (Evse evse : getStationStore().getEvses()) {
+            com.evbox.everon.ocpp.v20.message.common.Evse componentEvse = new com.evbox.everon.ocpp.v20.message.common.Evse()
+                    .withId(evse.getId());
+            for (Connector connector : evse.getConnectors()) {
+                // The evse is always available, the state for connector should be sent if different that the EVSE one
```

Comment:
didn't get the comment :)

Reply:
Because in the specification it says:

> For each Connector Component:
> - AvailabilityState (if known and different from EVSE).

And as the evse is always in state available, the report is added only if the connector is not available.
But correct me if i'm wrong.

Reply:
Evse is not always available as we do support ChangeAvailability case, please check the spec about it and adjust if necessary.